### PR TITLE
Move assumptions docs to a normal docs page

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
By moving them to a normal docs page we:
- Can add it to the sidebar
- Get the right sidebar with links to the various assumptions automatically
- Have `/docs` in the page URL

Preview: https://deploy-preview-2503--babel.netlify.app/docs/en/assumptions

### CODE REVIEW

File: website/static/js/components/assumption-repl.js [hunk 1]
```diff
@@ -1,32 +1,31 @@
...
-
-  <label><input id="enabled" type="checkbox" checked /> Enabled</label>
-  <mini-repl id="repl"></mini-repl>
+  <mini-repl id="repl" vertical></mini-repl>
 `;
 
-class AssumptionDocs extends HTMLElement {
-  _plugins = this.getAttribute("plugins").split(",");
-  _assumption = this.getAttribute("assumption");
+console.log("LOADING");
```
Comment (hzoo) [inline website/static/js/components/assumption-repl.js:10]: can remove?

File: website/static/js/components/assumption-repl.js [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
+  _plugins = this.dataset.plugins.split(",");
+  _assumption = this.dataset.assumption;
+  _input = this.querySelector("code.assumption-input").textContent;
 
   _enabled = true;
 
   constructor() {
     super();
 
+    console.log(this._input);
```
Comment (hzoo) [inline website/static/js/components/assumption-repl.js:22]: and here

File: docs/assumptions.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+<script type="module" src="/js/components/mini-repl.js"></script>
+<script type="module" src="/js/components/assumption-repl.js"></script>
+
+<style>
+.assumption-input {
+  display: none;
+}
+</style>
+
+## `arrayLikeIsIterable`
```
Comment (hzoo) [inline docs/assumptions.md:16]: What do you think about adding a short explanation? https://github.com/babel/website/pull/2502#discussion_r615103087 - maybe at least that this is added in 7.13 and it replaces what used to be loose

**Review by hzoo — APPROVED** [review]
🙌

File: website/static/js/components/assumption-repl.js (hunk 1 above)
Comment (nicolo-ribaudo) [reply website/static/js/components/assumption-repl.js:10]: oh right

**Pull request merged**